{
  log.info("StorePushTask.call() invoked for cluster URL: " + url);
  invokeHooks(BuildAndPushStatus.PUSHING,url);
  try {
    runPushStore(props,url,buildOutputDir);
    log.info("StorePushTask.call() finished for cluster URL: " + url);
    invokeHooks(BuildAndPushStatus.SWAPPED,url);
  }
 catch (  RecoverableFailedFetchException e) {
    log.warn("There was a problem with some of the fetches, but a swap was still able " + "to go through for cluster URL: " + url,e);
    invokeHooks(BuildAndPushStatus.SWAPPED_WITH_FAILURES,url);
  }
catch (  Exception e) {
    log.error("Exception during push for cluster URL: " + url + ". Rethrowing exception.");
    throw e;
  }
  return true;
}
